packages/parser/src/parser.ts

```diff
@@ -209,7 +209,12 @@ export function parseAndCompile(inText: string) {
 
             // Corresponds to the entity non-terminal in the grammar
             while (cursor.nextSibling()) {
-                if (cursor.name === "Text") {
+                if (
+                    cursor.name === "Text" ||
+                    cursor.name === "Ampersand" ||
+                    cursor.name === "EntityReference" ||
+                    cursor.name === "CharacterReference"
+                ) {
```

Comment:
`parser.ts` is only used in legacy code, right? The new DAST code doesn't touch it?

Reply:
There are other exports from `parser.ts` used in the new code, but they don't use the `parseAndCompile` export, which is the one changed in this PR. (If the had used `parseAndCompile`, they would have been losing any entities, etc.